Partial Sync SVGAnimationElement.idl with IDL Spec

Ahmad-S792 opened the pull request:
#### c7a661464e1fe2ba634157ac3b4ee756728be7a6
<pre>
Partial Sync SVGAnimationElement.idl with IDL Spec

<a href="https://bugs.webkit.org/show_bug.cgi?id=250575">https://bugs.webkit.org/show_bug.cgi?id=250575</a>
rdar://problem/104475517

Reviewed by Said Abou-Hallawa.

This patch is to align WebKit with web-spec [1].

[1]: <a href="https://svgwg.org/specs/animations/#InterfaceSVGAnimationElement">https://svgwg.org/specs/animations/#InterfaceSVGAnimationElement</a>

Cherry-Pick: <a href="https://src.chromium.org/viewvc/blink?revision=171485&amp">https://src.chromium.org/viewvc/blink?revision=171485&amp</a>;view=revision &amp;
<a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=190359

This patch aligns WebKit with web-specification by dropping unnecessary NaN
and remove &apos;optional&apos; and &apos;unrestricted&apos; from IDL file as needed.

* Source/WebCore/svg/SVGAnimationElement.cpp:
(SVGAnimationElement::beginElementAt): Remove early return for &apos;nan&apos; values
(SVGAnimationElement::endElementAt): Ditto
* Source/WebCore/svg/SVGAnimationElement.idl: Align with web-spec
* LayoutTests/svg/custom/elementTimeControl-nan-crash.html: Updated to Dump Text
* LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks.html: Add Testcase
* LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks-expected.txt: Add Testcase Expectation

Canonical link: <a href="https://commits.webkit.org/263253@main">https://commits.webkit.org/263253@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/812557bcc541a96cf786bdeee8c1c6ac41e2e137

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4018 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4115 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4227 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5461 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4267 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3996 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4207 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4090 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/4259 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4068 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4256 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3626 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5407 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1756 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3602 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/5732 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3581 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3663 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5132 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4070 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3275 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3585 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3602 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/989 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3637 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3864 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments.html:
```diff
@@ -0,0 +1,22 @@
+
+debug('');
+debug('beginElementAt(float offset)');
+shouldThrow('animateElm.beginElementAt()');
```
Please add this case to the SVGAnimationElement-begin-end-arguments-typechecks.html and remove this test.

shallawa commented on LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks.html:
```diff
@@ -0,0 +1,24 @@
+debug('beginElementAt(float offset)');
+shouldNotThrow('animateElm.beginElementAt(0)');
+shouldThrow('animateElm.beginElementAt(NaN)');
+shouldThrow('animateElm.beginElementAt(Infinity)');
```
Add this case here as well and get rid of the other test.
```
shouldThrow('animateElm.beginElementAt()');
````

Ahmad-S792 replied:
Done - added this and also in 'endElementAt' as well.

shallawa commented on LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks.html:
```diff
@@ -0,0 +1,24 @@
+<script>
+description('Check that invalid values of arguments throw TypeError.');
+
+var animateElm = document.querySelector('animate');
```
I think we do not like this kind of abbreviation for the variable name. Please use `animate`, `element` or `animateElement` instead.

Ahmad-S792 replied:
Done - changed to `animateElement`.

shallawa commented on LayoutTests/svg/custom/elementTimeControl-nan-crash.html:
```diff
@@ -2,14 +2,15 @@
         var svg = document.getElementById('svg');
-        animate.endElementAt(NaN);
-        animate.beginElementAt(NaN);
+        try { animate.endElementAt(NaN); } catch (e) {}
+        try { animate.beginElementAt(NaN); } catch (e) {}
```
I think it is better not to silence the exceptions. Please regenerate new expectations with the exceptions messages and keep this test as is.

Ahmad-S792 replied:
I did this change post your feedback in previous PR about this:
```Why do you want to make the expected file a dump render tree instead of dump as text? Also I do not see the changes in the test file LayoutTests/svg/custom/elementTimeControl-nan-crash.html to make it output this result.```

Ahmad-S792 replied:
@shallawa - Happy to revert and do platform specific expectation. I did this in my previous PR: [Link](https://github.com/WebKit/WebKit/pull/8660/files#diff-a9c09919ea260012874792863da1eba363af7acc7b5487c3d4e47e8aa790475f)

shallawa replied:
Your approach looks fine as well since this test was for a crash. So it is okay to ignore the exceptions.

Ahmad-S792 replied:
Thanks!

shallawa commented on LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks.html:
```diff
@@ -0,0 +1,24 @@
+debug('endElementAt(float offset)');
+shouldNotThrow('animateElm.endElementAt(0)');
+shouldThrow('animateElm.endElementAt(NaN)');
+shouldThrow('animateElm.endElementAt(Infinity)');
```
Ditto.

Ahmad-S792 replied:
Done. Used renamed variable.

Ahmad-S792 committed:
Partial Sync SVGAnimationElement.idl with IDL Spec

https://bugs.webkit.org/show_bug.cgi?id=250575
rdar://problem/104475517

Reviewed by Said Abou-Hallawa.

This patch is to align WebKit with web-spec [1].

[1]: https://svgwg.org/specs/animations/#InterfaceSVGAnimationElement

Cherry-Pick: https://src.chromium.org/viewvc/blink?revision=171485&view=revision &
https://src.chromium.org/viewvc/blink?view=revision&revision=190359

This patch aligns WebKit with web-specification by dropping unnecessary NaN
and remove 'optional' and 'unrestricted' from IDL file as needed.

* Source/WebCore/svg/SVGAnimationElement.cpp:
(SVGAnimationElement::beginElementAt): Remove early return for 'nan' values
(SVGAnimationElement::endElementAt): Ditto
* Source/WebCore/svg/SVGAnimationElement.idl: Align with web-spec
* LayoutTests/svg/custom/elementTimeControl-nan-crash.html: Updated to Dump Text
* LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks.html: Add Testcase
* LayoutTests/svg/dom/SVGAnimationElement-begin-end-arguments-typechecks-expected.txt: Add Testcase Expectation

Canonical link: https://commits.webkit.org/263253@main
